pkg/apis/hive/v1/clusterclaim_types.go

```diff
@@ -2,16 +2,17 @@ package v1
 
 import (
 	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ClusterClaimSpec defines the desired state of the ClusterClaim.
 type ClusterClaimSpec struct {
 	ClusterPoolName string `json:"clusterPoolName"`
 
-	// Subjects hold object references to authorize access to the claimed cluster.
+	// Subjects hold references to which to authorize access to the claimed cluster.
 	// +optional
-	Subjects []corev1.ObjectReference `json:"subjects,omitempty"`
+	Subjects []rbacv1.Subject `json:"subjects,omitempty"`
```

Comment:
How far have you tested this? I'd imagine we're most likely talking about OpenShift users here, but that concept does not exist in Kube IIRC. Will these Subjects be able to hold and work with users?

Reply:
Kube does have a concept of users. As far as I understand it, an OpenShift user is a Kube user. A subject can refer to a group, a user, or a service account.